app/classes/DB.php

```diff
@@ -406,23 +431,29 @@ protected function checkDataTable(){
      *
      * @param string $name The name of the data.
      * @param string $value The value of the data.
-     * @param bool $htmlspecialchars Whether to apply htmlspecialchars to the value or not.
+     * @param bool $htmlSpecialChars Whether to apply htmlspecialchars to the value or not.
+     *
+     * @throws Exception
      */
-    public function setData(string $name, string $value, bool $htmlspecialchars = false) {
+    public function setData(string $name, string $value, bool $htmlSpecialChars = false): void
+    {
         $this->checkDataTable();
         $query = "INSERT INTO `cml_data` (`data_name`, `data_value`, `data_created`)
                   VALUES ('$name', '$value', now())
-                  ON DUPLICATE KEY UPDATE `data_value` = " . ($htmlspecialchars ? '?' : "'$value'") . ", `data_lastModify` = now()";
-        $this->sql2db($query, $htmlspecialchars ? [$value] : []);
+                  ON DUPLICATE KEY UPDATE `data_value` = " . ($htmlSpecialChars ? '?' : "'$value'") . ", `data_lastModify` = now()";
+        $this->sql2db($query, $htmlSpecialChars ? [$value] : []);
     }
 
     /**
      * Retrieves data from the database based on the given name.
      *
      * @param string $name The name of the data to retrieve.
      * @return mixed The value of the data if found, or false if not found.
+     *
+     * @throws Exception
      */
-    public function getData(string $name){
+    public function getData(string $name): mixed
```

Comment:
The return type "mixed" is like the void, I don't like to type this to the function. But as I said, we can leave this if your happy with it